assets/blocks/jobs/types.jsx

```diff
@@ -0,0 +1,243 @@
+
+/**
+ * WordPress Dependencies.
+ */
+const { Component } = wp.element,
+	{ __ } = wp.i18n,
+	{ addFilter } = wp.hooks,
+	{
+		withAPIData,
+		CheckboxControl,
+		ToggleControl,
+	} = wp.components;
+
+/**
+ * External Dependencies.
+ */
+import _ from 'lodash';
+
+
+/**
+ * Job Types block attributes.
+ */
+const jobTypesAttributesConfig = {
+	showJobTypeFilters: {
+		type: 'boolean',
+		default: true,
+	},
+	includedJobTypes: {
+		type: 'string', // JSON object
+		default: '{}',
+	},
+	allJobTypes: {
+		type: 'string', // JSON array
+		default: '[]',
+	}
+};
+addFilter(
+	'wpjm_block_jobs_attributes_config',
+	'addJobTypesAttributes',
+	( attributesConfig ) => (
+		Object.assign( attributesConfig, jobTypesAttributesConfig )
+	)
+);
+
+
+/**
+ * Transform a job type attribute value from a JSON-encoded object to a
+ * comma-separated string.
+ *
+ * @param {string} value The JSON-encoded job type attribute value.
+ *
+ * @return {?string} Comma-separated value string for the attribute, or null.
+ */
+function transformJobTypeValue( value ) {
+	let obj = {};
+	try {
+		obj = JSON.parse( value );
+	} catch ( error ) {
+		// On error, just use the empty object.
```

Comment:
What is the result of value failing here? Does this really just result in `return null`? If so it may be better to more explicitly do that here.

Reply:
Yeah, it results in `return null`. I like that idea. Done.